File: 12_Local_Database_Connection_with_Docker/01_MongoDB_Connection_with_NodeJS/connect_database.js
const mongoose = require("mongoose");

// 👉 Replace this with your MongoDB connection string
// const MONGO_URI = "mongodb://localhost:27017/RAFC";
// const MONGO_URI = "mongodb://localhost:27017/RAFC" || "mongodb://host.docker.internal:27017/RAFC";
const MONGO_URI = process.env.MONGO_URI || "mongodb://host.docker.internal:27017/RAFC";

// 1️⃣ Connect to MongoDB
mongoose.connect(MONGO_URI)
.then(() => console.log("✅ Connected to RAFC database"))
.catch((err) => console.error("❌ DB connection error:", err));


// 2️⃣ Define schema for chats collection
const chatSchema = new mongoose.Schema({
    sender: String,
    message: String,
    timestamp: Date
}, { collection: "chats" });  
// 👆 very important: specify { collection: "chats" } to match existing collection


// 3️⃣ Create model
const Chat = mongoose.model("Chat", chatSchema);


// 4️⃣ Fetch all chat documents
async function fetchChats() {
    try {
        const chats = await Chat.find({});
        console.log("📄 Chats from RAFC DB:");
        console.log(chats);
    } catch (error) {
        console.error("❌ Error fetching chats:", error);
    } finally {
        mongoose.connection.close();  // close DB connection
    }
}

fetchChats();

File: 12_Local_Database_Connection_with_Docker/03_MongoDB_Connection_with_Python/connect_database.py
# pip install pymongo

from pymongo import MongoClient
import os
import sys

# 👉 Replace this with your MongoDB connection string or set MONGO_URI env var
MONGO_URI = os.environ.get("MONGO_URI", "mongodb://host.docker.internal:27017/RAFC")
# MONGO_URI = os.environ.get("MONGO_URI", "mongodb://localhost:27017/RAFC")

def connect_db(uri: str):
    try:
        client = MongoClient(uri, serverSelectionTimeoutMS=5000)
        # trigger server selection to validate connection
        client.server_info()
        print("✅ Connected to RAFC database")
        return client
    except Exception as e:
        print("❌ DB connection error:", e, file=sys.stderr)
        raise

def fetch_chats(db):
    try:
        chats = list(db["chats"].find({}))
        print("📄 Chats from RAFC DB:")
        for c in chats:
            print(c)
    except Exception as e:
        print("❌ Error fetching chats:", e, file=sys.stderr)
    finally:
        # close DB connection
        db.client.close()

if __name__ == "__main__":
    client = connect_db(MONGO_URI)
    # if MONGO_URI supplies a DB name (e.g. mongodb://.../RAFC), use get_default_database()
    try:
        db = client.get_default_database() or client["RAFC"]
    except Exception:
        # fallback to explicit database name
        db = client["RAFC"]

    fetch_chats(db)
